account_supplier_tax_rounding/models/account_tax.py

```diff
@@ -1,27 +1,36 @@
 # Copyright 2017 Graeme Gellatly
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
 
-from odoo import models, api
+from odoo import models
 
 
 class AccountTax(models.Model):
     _inherit = "account.tax"
 
-    @api.multi
     def compute_all(
-        self, price_unit, currency=None, quantity=1.0, product=None, partner=None
+        self,
+        price_unit,
+        currency=None,
+        quantity=1.0,
+        product=None,
+        partner=None,
+        is_refund=False,
+        handle_price_include=True,
     ):
         if (
             partner
             and partner.tax_calc_method == "round_per_line"
             and self.filtered(lambda t: t.type_tax_use == "purchase")
         ):
-            return super(AccountTax, self.with_context(round=True)).compute_all(
+            self = self.with_context(round=True)
+            return super().compute_all(
```

Comment:
This is wrong, you should be using the values passed, but anyway, you don't need this statement here at all. You can just remove the else clause and dedent the other super call.